src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/Mutect2FilteringEngine.java

```diff
@@ -187,12 +189,16 @@ private void applyGermlineVariantFilter(final M2FiltersArgumentCollection MTFAC,
         }
     }
 
-    private static void applyInsufficientEvidenceFilter(final M2FiltersArgumentCollection MTFAC, final VariantContext vc, final FilterResult filterResult) {
-        if (vc.hasAttribute(GATKVCFConstants.TUMOR_LOD_KEY)) {
-            final double[] tumorLods = getDoubleArrayAttribute(vc, GATKVCFConstants.TUMOR_LOD_KEY);
+    private void applyInsufficientEvidenceFilter(final M2FiltersArgumentCollection MTFAC, final VariantContext vc, final FilterResult filterResult) {
+        if (vc.hasAttribute(lodKey)) {
+            final double[] tumorLods = getDoubleArrayAttribute(vc, lodKey);
 
             if (MathUtils.arrayMax(tumorLods) < MTFAC.TUMOR_LOD_THRESHOLD) {
-                filterResult.addFilter(GATKVCFConstants.TUMOR_LOD_FILTER_NAME);
+                if(lodKey.equals(GATKVCFConstants.TUMOR_LOD_KEY)) {
```

Comment:
This is a code smell. I think Mitochondria pipeline should have its own filtering engine that inherits M2FilteringEngine

Reply:
I refactored, but still have this problem. It's only to get the correct filter name in the output. Is there a way to link the filter name with an annotation key?